doc/api/node_report.md

```diff
@@ -0,0 +1,198 @@
+# node-report
+
+Delivers a JSON-formatted Diagnostic summary, written to a file.
+
+The report is intended for development, test and production use, to capture
+and preserve information for problem determination. It includes JavaScript
+and native stack traces, heap statistics, platform information, resource
+usage etc. With the report option enabled, diagnostic reports can be triggered
+on unhandled exceptions, fatal errors and user signals, in addition to
+triggering programatically through API calls.
+
+The built-in report is supported in Node.js versions from REPLACEME onwards.
+For Node.js versions REPLACEME and below, use its [npm counterpart][] instead.
+
+## Usage
+
+```bash
+node --experimental-report --report-catch-exception
+ --report-catch-signal --report-catch-fatalerror app.js
+```
+
+* `--experimental-report` Enables the diagnostic report feature.
+ In the absence of this flag, all other report related options will be ignored.
```

Comment:
This might be surprising and take a bit to figure out, since it’s failing silently, unlike unknown options… I’d personally prefer to make the other flags imply `--experimental-report`, since they all represent a form of explicit opt-in, but I could see reluctancy from other collaborators…

Reply:
I'm not sure that --experimental-report makes sense to me because it then requires people to change the command line option when we promote to non-experimental.  I'd agree that just having the other three flags would be the right way to go, and to have them documented as experimental.

Reply:
If all other flags imply `--experimental-report`, it becomes redundant?

From an end-user's point of view, isn't it more consumable if we make it more explicit? For example:
 - `--experimental-report` only enables the feature, and exports the APIs
 - `--experimental-report` graduates to `--report` upon exiting experimental
 - `based on adoption and user feedback, `--report` may graduate to default in which case the flag disappears

does this proposition sounds reasonable to you all?

Reply:
@gireeshpunathil All I’m saying is that it’s not good for options to be ignored if the user expects them not to be ignored. I’d rather have us error out than fail silently.

> `--experimental-report` graduates to `--report` upon exiting experimental

What would that flag do then?

> based on adoption and user feedback, `--report` may graduate to default in which case the flag disappears

I think that’s a bit ambitious… Node.js doesn’t really generate any files on its own right now, and changing that is a pretty big modification to our UX.

Reply:
this is addressed now; `--experimental-report` acts like a lock to open up the feature, while other special flags enables report generation on respective events. Switching on special flags without the generic flag results in an error, so that usage is disambiguated.

Reply:
changes done to documentation as well.